File: client/src/models/Reminder/reminderDto.ts
import { RecurrencePatternDto } from "./recurrencePatternDto";

export interface ReminderDto {
  id: string;
  plantId: string;
  type: string;
  title: string;
  description: string;
  dueDate: string;
  recurrence?: RecurrencePatternDto | null;
  intensity: string;
  isCompleted: boolean;
  completedDate?: string | null;
  strategyId?: string | null;
  strategyParameters?: string | null;
  isStrategyOverride: boolean;
  createdAt: string;
  updatedAt?: string | null;
}

File: PlantCareBuddy/PlantCareBuddy.Application/DTOs/Reminder/ReminderDto.cs
using PlantCareBuddy.Domain.Enums;
using PlantCareBuddy.Domain.ValueObjects;

namespace PlantCareBuddy.Application.DTOs.Reminder
{
    public class ReminderDto
    {
        public Guid Id { get; set; }
        public Guid PlantId { get; set; }
        public string PlantName { get; set; }
        public ReminderType Type { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public DateTime DueDate { get; set; }
        public RecurrencePatternDto? Recurrence { get; set; }
        public bool IsCompleted { get; set; }
        public DateTime? CompletedDate { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
    }
}